cmd/snap/cmd_find_test.go

```diff
@@ -103,27 +95,43 @@ const findJSON = `
 
 func (s *SnapSuite) TestFind(c *check.C) {
 	n := 0
+	findWhat := []string{"hello", ""}
+
+	maxfindRequest := len(findWhat)
+
 	s.RedirectClientToTestServer(func(w http.ResponseWriter, r *http.Request) {
-		switch n {
-		case 0:
+		if n < maxfindRequest {
```

Comment:
The changes in this test function don't feel entirely sound. The changes generalized the logic, making it less readable, and didn't really add anything meaningful that would require that generalization. By the end of the day in both cases it simply sends a string that goes through the same code path.

If the goal is checking that the empty string does not fail, then let's simply invert the test that got removed (TestFindNothingFails => TestFindNothingWorks) instead of fiddling with this test function.